content/articles/2019-08-gargle-googledrive-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,91 @@
+
+The main change that users will notice is that gargle implements its own way of storing user OAuth tokens between sessions. The [httr package](https://httr.r-lib.org), which gargle depends on, has historically used a local `.httr-oauth` file for this. In contrast, gargle encourages tokens to be stored *outside* the project, in a hidden directory at the user level, in a key-value store that incorporates the Google user's email. This makes it less likely you'll push your tokens to the cloud, easier to use multiple identities in a project, and easier to share tokens across projects and packages. Users will notice they need to re-auth and may want to track down and delete vestigial `.httr-oauth` files.
+
+### Articles
+
+The release of gargle has provided an occasion to document several recurring workflows that can be tricky for useRs, which now work the same across multiple packages:
+
+  * [Non-interactive auth](https://gargle.r-lib.org/articles/non-interactive-auth.html)
+  * [Auth when using R in the browser](https://gargle.r-lib.org/articles/auth-from-web.html)
+  * [Managing tokens securely](https://gargle.r-lib.org/articles/articles/managing-tokens-securely.html)
+  * [How to get your own API credentials](https://gargle.r-lib.org/articles/get-api-credentials.html) 
+  
+Other articles are aimed at package maintainers interested in delegating auth or request & response handling to gargle:
+
+  * [How to use gargle for auth in a client package](https://gargle.r-lib.org/articles/gargle-auth-in-client-package.html)
+  * [Request helper functions](https://gargle.r-lib.org/articles/request-helper-functions.html)
+
+### Some gargle history
+
+I find gargle's origin story find very satisfying. I suspect that Hadley Wickham's bigrquery (first on CRAN January 2015) has shaped the basic design for how most R packages handle Google auth, directly or indirectly. I can certainly say that Joanna Zhao and I consulted it when developing googlesheets (first on CRAN July 2015). I know that Mark Edmondson was influenced, in turn, by googlesheets, when he developed googleAuthR (first on CRAN August 2015), used in his [suite of packages](https://code.markedmondson.me/r-packages/). By the time Lucy D'Agostino McGowan and I created googledrive (first on CRAN August 2017), I'd developed some strong opinions about how to modify httr's default behaviour for work with Google APIs. I was also working with Hadley Wickham at RStudio by that point and Lucy and I benefited from his design advice. In May-ish 2017, leading up to [rOpenSci's 2017 Unconf](https://unconf17.ropensci.org), Craig Citro [opened an issue](https://github.com/ropensci/unconf17/issues/85) to generate discussion about how to de-duplicate even more API pain, especially for Google APis. He has a wealth of experience from his role in shaping the [official Python client](https://developers.google.com/api-client-library/) and he made the initial commits that laid the foundation for `gargle::token_fetch()`.
```

Comment:
especially for Google APis. > especially for Google APIs.

Reply:
good catch, fixed